src/main/java/org/apache/sling/api/wrappers/CompositeValueMap.java

```diff
@@ -38,22 +43,36 @@
  * properties.
  * @since 2.3 (Sling API Bundle 2.5.0)
  */
+@ProviderType
 public class CompositeValueMap implements ValueMap {
 
-    /**
-     * Current properties
-     */
-    private final ValueMap properties;
+    private static final String IMMUTABLE_ERROR_MESSAGE = "CompositeValueMap is immutable";
+
+    private final Collection<ValueMap> valueMaps;
```

Comment:
should be a list, as it's ordered

Reply:
You're right! Thanks for picking that up! I had convinced myself that a Collection is an Iterable and therefore has an order - but of course even if it has a stable order (which it may not have), that is not a sufficient guarantee. We need to get exactly the order that we specified.